mne/cov.py

```diff
@@ -1595,6 +1565,8 @@ def regularize(cov, info, mag=0.1, grad=0.1, eeg=0.1, exclude='bads',
         regs['meg'] = mag
     else:
         regs.update(mag=mag, grad=grad)
+    if rank != 'full':
```

Comment:
idem. Why? can we avoid this branching?

Reply:
We can't easily avoid it in `regularize` because it has an additional (old) `proj` option